src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -202,38 +205,39 @@ bool CreaseDefenderFSM::ballNearbyWithoutThreat(const Update& event)
     Point robot_position = event.common.robot.position();
     Point ball_position  = event.common.world_ptr->ball().position();
 
-    std::optional<Robot> nearest_friendly =
+    std::optional<Robot> nearest_friendly_to_ball =
         event.common.world_ptr->friendlyTeam().getNearestRobot(ball_position);
-    std::optional<Robot> nearest_enemy =
+    std::optional<Robot> nearest_enemy_to_ball =
         event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
 
     if (event.control_params.ball_steal_mode == TbotsProto::BallStealMode::IGNORE)
     {
         // Do nothing if stealing is disabled
         return false;
     }
-    else if (nearest_friendly.has_value() &&
-             event.common.robot.id() != nearest_friendly.value().id())
+    else if (nearest_friendly_to_ball.has_value() &&
+             event.common.robot.id() != nearest_friendly_to_ball.value().id())
     {
-        // Do nothing if this robot is not the closest to the ball
+        // Do nothing if this robot is not the closest to the ball. Resolves issue of
+        // multiple simultaneous steals
         return false;
     }
-    else if (nearest_enemy.has_value())
+    else if (nearest_enemy_to_ball.has_value())
     {
-        double ball_distance =
-            distance(robot_position, event.common.world_ptr->ball().position());
-        double nearest_enemy_distance =
-            distance(robot_position, nearest_enemy->position());
+        double ball_distance_to_friendly = distance(robot_position, ball_position);
+        double ball_distance_to_enemy =
+            distance(nearest_enemy_to_ball.value().position(), ball_position);
 
         // Get the ball if the ball is on friendly side, nearby, and unguarded by the
         // enemy
         bool ball_is_near_friendly =
-            ball_distance < nearest_enemy_distance *
-                                crease_defender_config.max_get_ball_ratio_threshold();
+            ball_distance_to_friendly <
+            ball_distance_to_enemy *
+                (1.0 - crease_defender_config.max_get_ball_ratio_threshold());
```

Comment:
This change in the logic seems backwards to me. Shouldn't we be multiplying by `crease_defender_config.max_get_ball_ratio_threshold()` directly to get the ratio?

Reply:
You are right that mathematically speaking, it would be a direct multiplication. However, I think the meaning of `max_get_ball_ratio_threshold()` makes a lot more sense when we flip the percentage. For example, if `max_get_ball_ratio_threshold() = 0.1`, this means that the ball must be at least 10% closer to our defender than the enemy for stealing to occur.  Within the code though, this could be either represented by a multiplication of 0.9 to the enemy dist, or a multiplication of 1.1 to friendly dist. 

Below is some rambling since I changed the meaning of the math here today. It is only if you are curious why I made a change here.

I thought of an edge case and switched the logic to use this instead. The old code directly multiplied it since it was comparing the distance of the defender->ball vs nearest_enemy_to_defender->ball. This works in most circumstances, but in the case where the world looks like this:
Ball           Crease  Enemy
  x                 O       O
The crease would decide not to pursue the ball when it really should. Hence, I changed the logic to instead measure the distance from the ball to the nearest enemy and compare that with the current crease's distance. 

This means that I redefined the meaning of  `max_get_ball_ratio_threshold()`

```cpp
/* 
Max ratio between distances (crease defender and ball) / (nearest enemy and ball) for
pass defender to chase ball.
Threshold = 1.0 results in crease defenders chasing the ball even ball is equidistant
to nearest enemy. Threshold < 1.0 results in crease defenders chasing the ball only
when it is at least (threshold) percent closer to our bot than the nearest enemy.
*/
```

Reply:
I followed up in a new comment. Though Im still not sure if I fully understand the logic. I think I understand the edge case you're talking about, but I think in that case it's also okay to act as a regular crease defender trying to block the ball, as opposed to trying to take a risk by attempting to steal the ball.

Reply:
Also, regarding the comment:
> Threshold = 1.0 results in crease defenders chasing the ball even ball is equidistant to nearest enemy. Threshold < 1.0 results in crease defenders chasing the ball only when it is at least (threshold) percent closer to our bot than the nearest enemy.

If the value is set to 1, then right side of the equality becomes 0, meaning that the ball distance to friendly has to be less than 0 (impossible) for the friendly to try to steal the ball. That's why I think just multiplying makes more sense. Threshold of 1 should indeed mean that the crease defender is the most aggressive. And 0 be that the crease defender doesn't take any risks.